### PR TITLE
Fix recovery translog stats totals when recovering from store

### LINKED ISSUE #15974: _recovery api shows negative metrics after translog replay
Tested on 2.1.1:

```
      "translog" : {
        "recovered" : 13,
        "total" : -1,
        "percent" : "-1.0%",
        "total_on_start" : -1,
        "total_time_in_millis" : 340
      },
```

I have 13 entries in the translog for a shard, once it is done replaying on startup, if I hit the _recovery api and look at the translog section for the shard, it shows 13 recovered.  But the percent and total metrics are negative which is misleading.

### PR DESCRIPTION
**Pull request by bleskes**
Recovery from store fails to correctly set the translog recovery stats. This fixes it and tightens up the logic bringing it all to IndexShard (previously it was set by the recovery logic).

Closes #15974

### CODE REVIEW

**Comment by bleskes** [comment]
@s1monw can you take a look?

File: core/src/main/java/org/elasticsearch/index/shard/IndexShard.java [hunk 1]
```diff
@@ -874,6 +874,11 @@ public int performBatchRecovery(Iterable<Translog.Operation> operations) {
      * After the store has been recovered, we need to start the engine in order to apply operations
      */
     public void performTranslogRecovery(boolean indexExists) {
+        if (indexExists == false) {
+            // note: these are set when recovering from the translog
+            recoveryState().getTranslog().totalOperations(0);
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/index/shard/IndexShard.java]: any chance we can assign the `recoveryState().getTranslog()` somewhere and don't chain the calls all the time?

**Comment by s1monw** [comment]
LGTM left a cosmetic comment

**Comment by s1monw** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/index/shard/IndexShard.java (hunk 1 above)
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/index/shard/IndexShard.java]: I looked at it and tried but we only do it in 2 places and make the translog recovery state stats a thing will introduce unneeded bloot imh. I simplified things a (tiny) bit.

**Pull request closed without merging**